postboxes/views.py

```diff
@@ -0,0 +1,28 @@
+from django.db.models   import Q
+
+from rest_framework.generics   import ListCreateAPIView
+from rest_framework.filters    import SearchFilter, OrderingFilter
+from rest_framework.pagination import PageNumberPagination
+
+from postboxes.models      import Postbox
+from postboxes.serializers import PostboxListSerializer
+
+class PostboxListAPIView(ListCreateAPIView):
+    serializer_class = PostboxListSerializer
+
+    filter_backends = [SearchFilter, OrderingFilter]
+    search_fields   = ['name']
+    ordering_fields = ['send_at', 'closed_at', 'days_to_close', 'created_at']
+    ordering        = ['-created_at']
+
+    def get_queryset(self, *args, **kwargs):
+        status = self.request.GET.get('status') 
+
+        q = Q(days_to_close__gte=0)
+
+        if status == 'public':
+            q &= Q(is_public=True)
+        elif status == 'private':
+            q &= Q(is_public=False)
+
+        return Postbox.objects.filter(q) 
```

Comment:
처음 프로그램을 작성할때, 많은 분들이 로직으로만 코드를 작성하려고 하십니다.
이제는 이 생각의 틀을 깨셔야 합니다. 여기서 로직이라고 하는 의미는 if, else, for ...등을 사용해서 코드를 작성하는 것을 말합니다.
즉 절차지향 프로그래밍을 하는 것이죠.

if문을 사용하게 되면 `linear`하게 프로그래밍할 수 없고, if문 갯수만큼 경우의 수가 나눠지게 됩니다. 이렇게 되면 코드를 관리하기도 디버깅하기도 어려운 형태가 되버립니다.

그래서 최대한 linear하게 프로그래밍 할 수 있도록, 즉 구조적으로 프로그래밍 할 수 있도록 생각하시는게 좋아요~

아래 코드를 참고해보시면 이해하실 수 있을 거에요. 

```suggestion
        status = self.request.GET.get('status')
        
        status_set = {
            "public" : Q(is_public=True),
            "private" : Q(is_public=False)
        }
        
        return Postbox.objects.filter(status_set[status]) 
```
끝으로 위에서 설명드린 것과 Class가 탄생한 이유를 공부해보시면 많은 공부가 될 것 같습니다~